src/test/java/com/google/devtools/build/lib/remote/SimpleBlobStoreActionCacheTest.java

```diff
@@ -430,4 +440,71 @@ public void testDownloadFailsOnDigestMismatch() {
             () -> getFromFuture(client.downloadFile(fs.getPath("/exec/root/foo"), digest)));
     assertThat(e).hasMessageThat().contains(digest.getHash());
   }
+
+  private class ConcurrentMapBlobStore implements SimpleBlobStore {
```

Comment:
supernit: Maybe extract it into a separate file in test sources? Just to keep this file smaller :)

Could be package private if reusability is a concern